src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -0,0 +1,312 @@
+  if (!compareFilters(lastFilters ?? [], currentFilters ?? [], COMPARE_ALL_OPTIONS)) return false;
+  return true;
+};
+
+interface RangeSliderDataFetchProps {
+  fieldName: string;
+  dataViewId: string;
+  query?: ControlInput['query'];
+  filters?: ControlInput['filters'];
+}
+
+const fieldMissingError = (fieldName: string) =>
+  new Error(`field ${fieldName} not found in index pattern`);
+
+export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput, ControlOutput> {
+  public readonly type = RANGE_SLIDER_CONTROL;
+  public deferEmbeddableLoad = true;
+
+  private subscriptions: Subscription = new Subscription();
+  private node?: HTMLElement;
+
+  // Controls services
+  private dataService: ControlsDataService;
+  private dataViewsService: ControlsDataViewsService;
+
+  // Internal data fetching state for this input control.
+  private dataView?: DataView;
+  private field?: DataViewField;
+
+  // State to be passed down to component
+  private componentState: RangeSliderComponentState;
+  private componentStateSubject$ = new BehaviorSubject<RangeSliderComponentState>({
+    min: '',
+    max: '',
+    loading: true,
+    fieldFormatter: (value: string) => value,
+  });
+
+  constructor(input: RangeSliderEmbeddableInput, output: ControlOutput, parent?: IContainer) {
+    super(input, output, parent); // get filters for initial output...
+
+    // Destructure controls services
+    ({ data: this.dataService, dataViews: this.dataViewsService } = pluginServices.getServices());
+
+    this.componentState = {
+      min: '',
+      max: '',
+      loading: true,
+      fieldFormatter: (value: string) => value,
+    };
+    this.updateComponentState(this.componentState);
+
+    this.initialize();
+  }
+
+  private initialize = async () => {
+    const initialValue = this.getInput().value;
+    if (initialValue) {
+      await this.fetchMinMax();
+    }
+    this.setInitializationFinished();
+    this.setupSubscriptions();
+  };
+
+  private setupSubscriptions = () => {
+    const dataFetchPipe = this.getInput$().pipe(
+      map((newInput) => ({
+        lastReloadRequestTime: newInput.lastReloadRequestTime,
+        dataViewId: newInput.dataViewId,
+        fieldName: newInput.fieldName,
+        timeRange: newInput.timeRange,
+        filters: newInput.filters,
+        query: newInput.query,
+      })),
+      distinctUntilChanged(diffDataFetchProps)
+    );
+
+    // fetch available min/max when input changes or when search string has changed
+    this.subscriptions.add(dataFetchPipe.subscribe(this.fetchMinMax));
+
+    // build filters when value change
+    this.subscriptions.add(
+      this.getInput$()
+        .pipe(
+          debounceTime(400),
+          distinctUntilChanged((a, b) => isEqual(a.value, b.value)),
+          skip(1) // skip the first input update because initial filters will be built by initialize.
+        )
+        .subscribe(() => this.fetchMinMax())
```

Comment:
Is there a reason you're re-querying the min and max whenever the user makes a selection? This could potentially cause a lot of queries and slow down performance. If it's only in order to publish the filters, I would say maybe those two functions should be split up?